### PR TITLE
Callbacks refactored

### PR DESCRIPTION
**Pull request by philips77**
This PR changes the way callbacks are delivered by `BleManager`. The `BleManagerCallbacks` interface has been deprecated. All inner callbacks will still be called, but will be marked as deprecated and will be removed in some point in the future.

Instead, 2 new methods have been added to `BleManager`:
1. `setDisconnectCallback` - an optional callback for receiving disconnection events with disconnection reason. It's already available in 2.2.0-alpha07, but the behavior has been modified a bit. It's called in a different moment, and isn't called when the lib is retrying to connect.
2. `setBondingCallback`- an optional callback to receive events related to device bonding state.

Other callbacks should be handler on pre-request basis. For example, instead of `onDeviceReady`, you should set `.done(SuccessCallback)` callback on the `ConnectRequest`. Some callbacks have been deprecated without providing any replacement. For example, `onServicesDiscovered` was called before the initialization queue starts to be executed. But in most cases, users are interested in when the device is ready, not when it's connected. Also, `onError` will be removed, as there are failure callbacks for all the callbacks.

### CODE REVIEW

**Comment by philips77** [comment]
The PR changes API implemented in #149, related to #110 and #21.

**Comment by philips77** [comment]
@moritz-eder @sssemil

File: README.md [hunk 1]
```diff
@@ -89,7 +93,7 @@ The first step is to create your BLE Manager implementation, like below. The man
 act as API of your remote device, to separate lower BLE layer from the application layer.
 ```java
 
-class MyBleManager extends BleManager<BleManagerCallbacks> {
+class MyBleManager extends BleManage {
```
Comment (moritz-eder) [inline README.md:96]: typo: BleManager

File: README.md [hunk 2]
```diff
@@ -244,7 +248,7 @@ To connect to a Bluetooth LE device using GATT, create a manager instance:
 
 ```java
 final MyBleManager manager = new MyBleManager(context);
-manager.setManagerCallbacks(fluxCallbacks);
+manager.setDisconnectCallbacks(abortCallback);
```
Comment (moritz-eder) [inline README.md:251]: typo: setDisconnectCallback

**Comment by philips77** [comment]
Fixed. Also I renamed `BleServerManagerCallbacks` to `ServerCallback` and moved it to `callback` package.

**Pull request merged**